Sources/Task/TaskRecover.swift

```diff
@@ -47,4 +47,18 @@ extension Task {
         return Task<SuccessValue>(future: future, cancellation: cancel)
         #endif
     }
+
+    public func recoverWithNewTask(upon executor: Executor, start startNextTask: @escaping(Error) -> Task<SuccessValue>) -> Task<SuccessValue> {
+        
+        let future: Future<TaskResult<SuccessValue>> = andThen(upon: executor) { (result) -> Task<SuccessValue> in
+            do {
+                let value = try result.extract()
```

Comment:
As I read it, this puts more weight on the extraction than the wrapping, but I'm not sure that's where we want the focus to go. I'd inline it as `return Task<SuccessValue>(success: try result.extract())`.